lib/version_manifest.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+class VersionManifest
+  attr_reader :gem, :version, :fs, :gem_content
+
+  delegate :gem, :fs, to: :gem_content
+
+  def initialize(gem:, version:, fs: GemContent.fs)
+    raise ArgumentError, "version must be a valid version name" unless version.try(:match?, Rubygem::NAME_PATTERN)
```

Comment:
why would the version match the gem name pattern?

Reply:
We use the same name pattern for both rubygem names and [version numbers and platforms (AKA `slug`, which is what I'm expecting here)](https://github.com/rubygems/rubygems.org/blob/master/app/models/version.rb#L23-L24). This is more of a sanity check to make sure you don't pass in the object since it's just called `version`. I didn't like naming it `version_slug` or something longer since it just made this whole class uglier.

Reply:
I updated this to use number/platform because that saves some complexity in a real API.